RocketChat/Rocket.Chat
# [NEW] Making connector as a part of VoipService and removing earlier hardcoding for management server

Clickup Task : https://app.clickup.com/t/7qeq76
Description :
Some background :
Issue#1 : CommandHandler class is an entry point to a connector to Asterisk. This command handler had hardcoded values till the APIs and database for the management API was getting ready.
Once it got ready there was a need to change it and use the database values. This design change is triggered by this need.
The aim for the re-design was that all the API access should happen via voip service. It was realised that |CommandHandler| gets created (Because it is declared globally in REST APIs) before the Voip service gets initialised. And because of this fact, CommandHandler does not get to read the values as the service has not yet started and initialised |VoipServerConfiguration|.

To fix this issue, |CommandHandler| has to be created after service and should be accessible only via service. So it has been moved to Voip service. Few more points to consider here is that
CommandHandler::initConnection may not work always. When Voip is getting used for the first time, the server values (management and callserver) will be empty. One has to add those values using the admin interface. So CommandHandler::initConnection failure should not cause server to crash. So errors from CommandHandler::initConnection should be written in logs and the code should move ahead.

Issue#2 : Some design refinements have been done. The intelligence in the REST APIs have been reduced. While building the code connector was exposed outside. Now connector is contained within the
service. Service contains all necessary implementation. The necessary changes have been done to support this architectural change.







## Proposed changes (including videos or screenshots)

Considering these points, following changes have been done.
1. Voip rest APIs which use CommandHandler (Queue and extension APIs) now query for the CommandHandler instead of creating it.
2. server-config.ts REST API for adding management interface reinitialises the connection after adding a management interface.
3. On the server side, Voip Service interface has been changed, to have new methods. Voip service and CommandHandler is changed to adapt to the design mentioned above.
4. Log level change in AMIConnection file.
5. Modified the IVoipService interface to contain all the necessary methods and changed service.ts to have the implementation.



## Issue(s)


## Steps to test or reproduce
This should be tested in couple of configurations :
1. When none of the servers (Call-server and Management server) are configured. Expectation : Server should start and allow configuring call server and management server using the REST APIs/Admin interface.
2. When either one of them is configured. Expectation : Server should start and remaining server should be configurable using the REST APIs/Admin interface.
3. When both are configured. Expectation : Endpoint should get necessary information for registration. Other management/Admin APIs should work.
4. When management server values are changed. Expectation : Reinitialisation of connector happens and the information is pulled from new management server. Verify this by having 2 asterisk servers with different configuration.

## Further comments
